File: examples/trust-gate.js
// Cognitive Passport — trust gate (JavaScript). No SDK needed: the contract is the API.
//
// The gate is FAIL-CLOSED: it denies unless the passport positively clears every check.
// `edge_proven: null` (ahead but not statistically confirmed) and `false` (not proven) BOTH deny.

const REFERENCE = "https://metainsight.app"; // reference provider; pass your own baseUrl to read another

async function getPassport(agentId, baseUrl = REFERENCE) {
  const r = await fetch(`${baseUrl}/passport/${agentId}?format=json`);
  if (!r.ok) throw new Error("passport unavailable");
  return r.json();
}

// Decide whether to delegate an action to an agent, based on its verifiable history.
// Evidence lives in the passport; POLICY (these thresholds) is yours — tune them.
async function canDelegate(agentId, {
  baseUrl = REFERENCE,
  minResolved = 10,        // require resolved (not merely committed) history
  domain,                  // optional: require a track record in this specific domain
  requireVerified = false, // set true for high-risk actions (money, code, permissions)
} = {}) {
  const p = await getPassport(agentId, baseUrl);

  if (p.edge_proven !== true)
    return { ok: false, why: "edge not proven" };                 // null AND false both deny

  if ((p.resolved_forecasts ?? 0) < minResolved)
    return { ok: false, why: "insufficient resolved history" };   // resolved, not just committed

  if (domain) {
    const d = (p.domains || []).find((x) => x.domain === domain);
    if (!d || (d.n ?? 0) < minResolved)
      return { ok: false, why: `insufficient track record in ${domain}` };
  }

  if (requireVerified && p.verified !== true)
    return { ok: false, why: "evidence not independently verified" };

  return { ok: true, calibration: p.calibration };
}

// Before handing an agent capital or permissions:
const gate = await canDelegate("noesis", { minResolved: 10, requireVerified: true });
if (!gate.ok) console.log("do NOT delegate:", gate.why);
else console.log("ok — calibration:", gate.calibration);
// Note: with a strict gate, none of the reference agents pass yet — none has a *proven* edge.
// That is the point: an honest trust layer refuses to green-light the unproven, incl. its author's own agents.

File: examples/trust_gate.py
"""Cognitive Passport — trust gate (Python). The contract is the API; no package required.

The gate is FAIL-CLOSED: it denies unless the passport positively clears every check.
`edge_proven` of None (ahead but unconfirmed) and False (not proven) BOTH deny.
"""
import requests

REFERENCE = "https://metainsight.app"  # reference provider; pass your own base_url to read another


def get_passport(agent_id: str, base_url: str = REFERENCE) -> dict:
    r = requests.get(f"{base_url}/passport/{agent_id}", params={"format": "json"})
    r.raise_for_status()
    return r.json()


def can_delegate(
    agent_id: str,
    base_url: str = REFERENCE,
    min_resolved: int = 10,          # require resolved (not merely committed) history
    domain: str | None = None,       # optional: require a track record in this domain
    require_verified: bool = False,  # True for high-risk actions (money, code, permissions)
) -> tuple[bool, str]:
    p = get_passport(agent_id, base_url)

    if p.get("edge_proven") is not True:               # None and False both deny
        return False, "edge not proven"
    if (p.get("resolved_forecasts") or 0) < min_resolved:
        return False, "insufficient resolved history"
    if domain:
        d = next((x for x in p.get("domains", []) if x["domain"] == domain), None)
        if not d or (d.get("n") or 0) < min_resolved:
            return False, f"insufficient track record in {domain}"
    if require_verified and p.get("verified") is not True:
        return False, "evidence not independently verified"
    return True, "ok"


if __name__ == "__main__":
    ok, why = can_delegate("trader", require_verified=True)
    print("delegate" if ok else f"do NOT delegate — {why}")
    # With a strict gate, none of the reference agents pass yet — none has a *proven* edge.
    # That is the point: an honest trust layer refuses to green-light the unproven.
